### ISSUE TITLE
[theme] Wrong coloring in for cycle and break whole coloring

### ISSUE THREAD

**Issue by aeschli** [Microsoft/TypeScript-TmLanguage#215]
_From @zozzz on July 15, 2016 13:19_

- VSCode Version: 1.3.1
- OS Version: Windows 10

Steps to Reproduce:

1. Make a javascript file with the following contents:
```javascript
function test() {
	for (var i=0 ; i<l ; ++i) {

	}
}
```
another example:
```javascript
for (var i=0 ; i<l ; ++i) {

}
```
![example](https://cdn.pbrd.co/images/bMlkBeGSd.jpg)

_Copied from original issue: Microsoft/vscode#9341_

**Comment by aeschli** [comment]
Note that we already use the changes from https://github.com/Microsoft/TypeScript-TmLanguage/pull/191 for JavaScript

**Comment by sheetalkamat** [comment]
Fixed by #257
